node/src/components/storage.rs

```diff
@@ -1329,7 +1340,7 @@ impl Storage {
     }
 
     // Retrieves a block header by hash.
-    pub(crate) fn get_block_header_by_hash(
+    pub(crate) fn read_block_header_by_hash(
```

Comment:
Is this rename unnecessary? I don't understand why you're making these decision unilaterally. There's `get_single_block`, `get_blocks_while`, `get_block_body_for_*`, etc.

Reply:
Over time, the convention in this file has become that `get_` methods are private and take a transaction and `read_` methods are public and do not take a transaction.

Reply:
Thanks for the explanation. This is a weird design decision. I don't see any advantage of it - no one will memorize that rule anyway. Developers will type in `get_` or `read_` and then look for what the IDE is suggesting. It's easier to remember one rule (`get_` or `read_`) than thinking whether I have a transaction or not and whether I want to call public or private endpoint.